LLVMFloatToFixed/IndirectCallConversion.cpp

```diff
@@ -0,0 +1,96 @@
+#include "LLVMFloatToFixedPass.h"
+#include "TypeUtils.h"
+#include "llvm/ADT/SmallPtrSet.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Transforms/Utils/Cloning.h"
+#include "llvm/Transforms/Utils/ValueMapper.h"
+
+using namespace llvm;
+using namespace taffo;
+using namespace flttofix;
+
+using handler_function = void (FloatToFixed::*)(
+    llvm::CallInst *patchedDirectCall, llvm::Function *indirectFunction);
+
+/// Map to keep track of the handled indirect functions for the conversion.
+const std::map<const std::string, handler_function> indirectCallFunctions = {
+    {"__kmpc_fork_call", &FloatToFixed::handleKmpcFork}};
+
+/// Retrieve the indirect calls converted into trampolines and re-use the
+/// original indirect functions.
+void FloatToFixed::convertIndirectCalls(llvm::Module &m) {
+  std::vector<llvm::CallInst *> trampolineCalls;
+
+  // Retrieve the trampoline calls using the INDIRECT_METADATA
+  for (llvm::Function &curFunction : m) {
+    for (auto instructionIt = inst_begin(curFunction);
+         instructionIt != inst_end(curFunction); instructionIt++) {
+      if (auto curCallInstruction =
+              llvm::dyn_cast<llvm::CallInst>(&(*instructionIt))) {
+        if (curCallInstruction->getMetadata(INDIRECT_METADATA)) {
+          trampolineCalls.push_back(curCallInstruction);
+        }
+      }
+    }
+  }
```

Comment:
In theory you should have done this before the various convertXXX methods are called in a separate pass, but that's just for consistency with the existing structure of the code.

And considering that the existing structure of conversion is far from rock solid (on the contrary it's frail like a pudding) who cares if you deviate from the existing patterns

Reply:
It's a while that we don't touch it, but if I remember correctly the retrieval of the trampoline calls should be done **after** the various conversion, so that TAFFO can convert the trampoline functions without seeing any OpenMP runtime library function. 

After the conversion, we retrieve the direct and converted (or not, in case TAFFO doesn't convert) trampoline functions and substitute them with the indirect ones.